### DISCUSSION TITLE
How include external package in parcel bundle ?

### DISCUSSION THREAD

**Discussion by zababurinsv** [parcel-bundler/parcel#7046] [Q&A] [answered]
# 🐛 bug report

I have two projects  
* Project №1  -  build with parcel
* Add this bundle in project №2
* Project №2  -  build with parcel

## 🎛 Configuration (.babelrc, package.json, cli command)

* `package.json`
```js
  "devDependencies": {
    "@babel/preset-react": "^7.14.5",
    "npm-run-all": "^4.1.5",
    "parcel": "2.0.0-rc.0",
    "parcel-reporter-static-files-copy": "^1.3.0"
  },
  "scripts": {
    "start": "start",
    "dev": "parcel ./src/index.html -p $npm_package_config_port --cache-dir ./.parcel-cache",
    "build": "parcel build ./src/index.mjs --cache-dir ./.parcel-cache ",
    "kill": "kill $(lsof -t -i:$npm_package_config_port)",
    "clean": "rimraf ../../.parcel-cache",
    "compress": "terser ./dist/index.mjs -o ./dist/index.min.mjs -c -m"
  },
```
* `.parcelrc`  

```js
{
  "extends": "@parcel/config-default",
   "reporters":  ["...", "parcel-reporter-static-files-copy"]
}
```

## 🤔 Expected Behavior

Then i added external package  
```
import parser from "ua-parser-js";
```
Build Project №1 give me `import $eoIL6$uaparserjs from "ua-parser-js";`
It is module not include in bundle.

And then i build Project №2 
I get error

```
🚨 Build failed.

@parcel/core: Failed to resolve 'ua-parser-js' from './src/github.com/zababurinsv/test_07_10_2021/docs/index.mjs'

  /home/zb/conceptnet5/src/github.com/zababurinsv/test_07_10_2021/docs/index.mjs:1:31
  > 1 | import $eoIL6$uaparserjs from "ua-parser-js";
  >   |                               ^
    2 | 
    3 | function $parcel$interopDefault(a) {

@parcel/resolver-default: External dependency "ua-parser-js" is not declared in package.json.

```

## 💁 Possible Solution

Include external package in result bundle.
Maybe make a separate option?
And that would result in one file.
This is necessary in order to collect one main bundle from many small projects.

## 💻 Code Sample

Project №1

```js
import parser from "ua-parser-js";

export let test = async ( path = false, devTool = true ) => {
    try {
        console.log({
            type: parser
        })
   }
```
Project №2
import Project№1 from './index.mjs'

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | "2.0.0-rc.0",
| Node             | v16.9.1
| npm/Yarn         | Yarn  1.22.11
| Operating System | Ubuntu 20.04

**Comment by DeMoorJasper** [comment, marked as answer]
There's an option to include node_modules https://parceljs.org/features/targets/#includenodemodules
